website/docs/flow-sensitive.md

```diff
@@ -96,6 +96,63 @@ The complete list of constructs that affect Sorbet's flow-sensitive typing:
 > overridden! For example, Sorbet can behave unpredictably when overriding
 > `is_a?` in weird ways.
 
+## What about `respond_to?`?
+
+Sorbet cannot support flow sensitivity for `respond_to?` in the way that most
+people expect.
+
+For example:
+
+```ruby
+sig {params(x: Object).void}
+def flow_sensitivity(x)
+  # Does not work:
+  if x.respond_to?(:foo)
+    T.reveal_type(x) # => Object
+    x.foo # Method `foo` does not exist
+  end
+end
+```
+
+In this example, knowing that `x` responds to a method with the name `foo` does
+not tell Sorbet anything more specific about the type for `x`. Sorbet does not
+have any sort of duck-typed interfaces that let Sorbet update it's knowledge of
+the type of `x` to, "`Object` plus responds to `foo`" so it must keep the type
+of `x` at `Object`, which does not allow calling `foo`.
+
+Note that even if Sorbet did support such a type, it's likely that `respond_to?`
+would _still_ not be supported, because knowing that the method `foo` exists
```

Comment:
Is "still not be supported" clearly referring to support for flow sensitivity in this context?  I can see the argument for, but it still feels slightly jarring to read.

Reply:
clarified